include/oneapi/dpl/experimental/kt/internal/esimd_radix_sort_dispatchers.h

```diff
@@ -266,33 +243,132 @@ __onesweep(sycl::queue __q, _RngPack&& __pack, ::std::size_t __n)
     constexpr ::std::uint32_t __hist_work_group_size = 64;
     __event_chain = __radix_sort_histogram_submitter<__is_ascending, __radix_bits, __hist_work_group_count,
                                                      __hist_work_group_size, _EsimdRadixSortHistogram>()(
-        __q, __pack.__keys_rng(), __mem_holder.__global_hist_ptr(), __n, __event_chain);
+        __q, __input_pack.__keys_rng(), __mem_holder.__global_hist_ptr(), __n, __event_chain);
 
     __event_chain = __radix_sort_onesweep_scan_submitter<__stage_count, __bin_count, _EsimdRadixSortScan>()(
         __q, __mem_holder.__global_hist_ptr(), __n, __event_chain);
 
-    for (::std::uint32_t __stage = 0; __stage < __stage_count; __stage++)
+    auto __submit_iteration = [&](auto __input_, auto __output_, auto __p_global_hist_, auto __p_group_hists_,
```

Comment:
My intention is to reduce repetitive code to make more apparent what is actually different between these invocations of `__radix_sort_onesweep_submitter`, but I don't think it is required technically.  

Also, my changes unroll the first iteration of the submission before the loop, as its inputs / outputs are special cased so there are three redundant calls, rather than just two.

I can go back to something more similar to the previous way of doing it without the lambda.  I don't have a strong preference here.

Reply:
I've removed the lambda `__submit_iteration`.